lms/djangoapps/instructor/enrollment.py

```diff
@@ -229,20 +229,18 @@ def reset_student_attempts(course_id, student, module_state_key, delete_module=F
                 except StudentModule.DoesNotExist:
                     # If a particular child doesn't have any state, no big deal, as long as the parent does.
                     pass
+        # Some blocks (openassessment) use StudentModule data as a key for internal submission data.
+        # Inform these blocks of the reset and allow them to handle their data.
+        clear_submission = getattr(block, "clear_submission", None)
+        if callable(clear_submission):
+            clear_submission(
+                user_id=user_id,
+                course_id=course_id.to_deprecated_string(),
+                item_id=module_state_key.to_deprecated_string()
+            )
     except ItemNotFoundError:
         log.warning("Could not find %s in modulestore when attempting to reset attempts.", module_state_key)
 
-    # Reset the student's score in the submissions API
-    # Currently this is used only by open assessment (ORA 2)
-    # We need to do this *before* retrieving the `StudentModule` model,
-    # because it's possible for a score to exist even if no student module exists.
-    if delete_module:
-        sub_api.reset_score(
-            anonymous_id_for_user(student, course_id),
-            course_id.to_deprecated_string(),
-            module_state_key.to_deprecated_string(),
-        )
```

Comment:
ORA2 is not the only thing that uses submissions, though the number of third party blocks that do is limited. Submissions was meant to be an API usable by other XBlocks, for storing more easily audited submission and scoring information, especially across users.